akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2229,6 +2229,98 @@ private[stream] object Collect {
 
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] final class StatefulMapConcatWithContext[In, Ctx, Out](
```

Comment:
This is mostly a copy of `StatefulMapConcat`.  Maybe they could be merged, if it makes sense.

Reply:
Couldn't it just be using `statefulMapConcat`?

Reply:
It could, but I didn't update `statefulMapConcat` to use the new custom stage.  A new stage is required to execute the UDFs at the right time.